test/integration/controller/jobset/jobset_controller_test.go

```diff
@@ -0,0 +1,667 @@
+		parallelism := jobSet.Spec.ReplicatedJobs[0].Replicas
+		newParallelism := parallelism + 1
+		createdJobSet.Spec.ReplicatedJobs[0].Replicas = newParallelism
+		gomega.Expect(k8sClient.Update(ctx, createdJobSet)).Should(gomega.Succeed())
+		gomega.Eventually(func() bool {
+			if err := k8sClient.Get(ctx, lookupKey, createdJobSet); err != nil {
+				return false
+			}
+			return createdJobSet.Spec.Suspend != nil && *createdJobSet.Spec.Suspend &&
+				len(jobSet.Spec.ReplicatedJobs[0].Template.Spec.Template.Spec.NodeSelector) == 0
+		}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+		gomega.Eventually(func() bool {
+			ok, _ := testing.CheckLatestEvent(ctx, k8sClient, "DeletedWorkload", corev1.EventTypeNormal, fmt.Sprintf("Deleted not matching Workload: %v", wlLookupKey.String()))
+			return ok
+		}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+
+		ginkgo.By("checking the workload is updated with new count")
+		gomega.Eventually(func() bool {
+			if err := k8sClient.Get(ctx, wlLookupKey, createdWorkload); err != nil {
+				return false
+			}
+			return createdWorkload.Spec.PodSets[0].Count == int32(newParallelism)
+		}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+		gomega.Expect(createdWorkload.Status.Admission).Should(gomega.BeNil())
+
+		ginkgo.By("checking the JobSet is unsuspended and selectors added when workload is assigned again")
+		admission = testing.MakeAdmission(clusterQueue.Name).
+			PodSets(
+				kueue.PodSetAssignment{
+					Name: "replicated-job-1",
+					Flavors: map[corev1.ResourceName]kueue.ResourceFlavorReference{
+						corev1.ResourceCPU: "on-demand",
+					},
+					Count: pointer.Int32(createdWorkload.Spec.PodSets[0].Count),
+				},
+				kueue.PodSetAssignment{
+					Name: "replicated-job-2",
+					Flavors: map[corev1.ResourceName]kueue.ResourceFlavorReference{
+						corev1.ResourceCPU: "spot",
+					},
+					Count: pointer.Int32(createdWorkload.Spec.PodSets[1].Count),
+				},
+			).
+			Obj()
+		gomega.Expect(util.SetAdmission(ctx, k8sClient, createdWorkload, admission)).Should(gomega.Succeed())
+		gomega.Eventually(func() bool {
+			if err := k8sClient.Get(ctx, lookupKey, createdJobSet); err != nil {
+				return false
+			}
+			return !*createdJobSet.Spec.Suspend
+		}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+
+		gomega.Expect(len(createdJobSet.Spec.ReplicatedJobs[0].Template.Spec.Template.Spec.NodeSelector)).Should(gomega.Equal(1))
+		gomega.Expect(createdJobSet.Spec.ReplicatedJobs[0].Template.Spec.Template.Spec.NodeSelector[labelKey]).Should(gomega.Equal(onDemandFlavor.Name))
+		gomega.Expect(len(createdJobSet.Spec.ReplicatedJobs[1].Template.Spec.Template.Spec.NodeSelector)).Should(gomega.Equal(1))
+		gomega.Expect(createdJobSet.Spec.ReplicatedJobs[1].Template.Spec.Template.Spec.NodeSelector[labelKey]).Should(gomega.Equal(spotFlavor.Name))
+
+		gomega.Consistently(func() bool {
```

Comment:
Let's remove this test, to make it faster. We have sufficient coverage for similar behavior for job.

Reply:
done